watertap/examples/flowsheets/oaro/oaro.py

```diff
@@ -0,0 +1,480 @@
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+import os
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    Constraint,
+    Expression,
+    Objective,
+    Var,
+    Param,
+    NonNegativeReals,
+    TransformationFactory,
+    units as pyunits,
+    assert_optimal_termination,
+    check_optimal_termination,
+)
+from pyomo.network import Arc, SequentialDecomposition
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers import get_solver
+from idaes.core.util.exceptions import InitializationError
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import (
+    solve_indexed_blocks,
+    propagate_state,
+)
+from idaes.models.unit_models import Mixer, Separator, Product, Feed
+from idaes.models.unit_models.mixer import MomentumMixingType
+from idaes.core import UnitModelCostingBlock
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+from idaes.core.util.misc import StrEnum
+
+import watertap.property_models.NaCl_prop_pack as props
+from watertap.unit_models.reverse_osmosis_0D import (
+    ReverseOsmosis0D,
+    ConcentrationPolarizationType,
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+from watertap.unit_models.osmotically_assisted_reverse_osmosis_0D import (
+    OsmoticallyAssistedReverseOsmosis0D,
+    # ConcentrationPolarizationType,
+    MassTransferCoefficient,
+    PressureChangeType,
+)
+from watertap.unit_models.pressure_changer import Pump, EnergyRecoveryDevice
+from watertap.core.util.initialization import assert_degrees_of_freedom
+from watertap.costing import WaterTAPCosting
+
+
+class ERDtype(StrEnum):
+    pump_as_turbine = "pump_as_turbine"
+
+
+def erd_type_not_found(erd_type):
+    raise NotImplementedError(
+        "erd_type was {}, but can only " "be pump_as_turbine" "".format(erd_type.value)
+    )
+
+
+def main(erd_type=ERDtype.pump_as_turbine):
+    # set up solver
+    solver = get_solver()
+
+    # build, set, and initialize
+    m = build(erd_type=erd_type)
+    set_operating_conditions(m)
+    initialize_system(m, solver=solver)
+
+    results = solve(m, solver=solver)
+
+    print("\n***---Simulation results---***")
+    display_system(m)
+    if erd_type == ERDtype.pump_as_turbine:
+        display_state(m)
+    else:
+        pass
+
+    return m
+
+
+def build(erd_type=ERDtype.pump_as_turbine):
+    # TODO: add costing later (OARO unit model does not have costing method)
+
+    # flowsheet set up
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.erd_type = erd_type
+    m.fs.properties = props.NaClParameterBlock()
+    # m.fs.costing = WaterTAPCosting()
+
+    # Control volume flow blocks
+    m.fs.feed = Feed(property_package=m.fs.properties)
+    m.fs.product = Product(property_package=m.fs.properties)
+    m.fs.product2 = Product(property_package=m.fs.properties)
```

Comment:
```suggestion
    m.fs.product2 = Product(property_package=m.fs.properties)
```
This can be deleted now. I originally added this when we were playing around with constructing the flowsheet one unit at a time.

Reply:
Deleted